src/taraxad/main.cpp

```diff
@@ -10,60 +14,533 @@ using namespace std;
-    bpo::options_description allowed_options("Allowed options");
-    allowed_options.add(main_options);
+        "database tables - this could take a long time")("rebuild_db_period",
+                                                         bpo::value<uint64_t>(&rebuild_db_period)->default_value(0),
+                                                         "Use with rebuild-db - Rebuild db up to a specified period")(
+        "rebuild-network", bpo::bool_switch(&rebuild_network)->default_value(false),
+        "Delete all saved network/nodes information and rebuild network from boot nodes")(
+        "revert-to-period", bpo::value<uint64_t>(&revert_to_period)->default_value(0),
+        "Revert db/state to specified period (specify period)");
+
+    bpo::options_description allowed_options;
+    allowed_options.add(help_option);
+    allowed_options.add(node_options);
     bpo::variables_map option_vars;
     auto parsed_line = bpo::parse_command_line(argc, argv, allowed_options);
     bpo::store(parsed_line, option_vars);
     bpo::notify(option_vars);
+
     if (option_vars.count("help")) {
-      cout << allowed_options << endl;
+      nodeInfo();
+      cout << help_option << endl;
+      commandsInfo();
+      cout << node_options << endl;
+
       return 1;
     }
-    if (option_vars.count("version")) {
-      cout << TARAXAD_VERSION << endl;
-      return 1;
+
+    // Commands
+    bpo::options_description commands("COMMANDS");
+    commands.add_options()("command", bpo::value<string>(), "command to execute")("subargs", bpo::value<string>(),
+                                                                                  "Arguments for command");
+    bpo::positional_options_description command_pos;
+    command_pos.add("command", 1).add("subargs", 2);
+    bpo::parsed_options command_parsed =
+        bpo::command_line_parser(argc, argv).options(commands).positional(command_pos).allow_unregistered().run();
+    bpo::variables_map command_vm;
+    bpo::store(command_parsed, command_vm);
+    if (!command_vm.empty()) {
+      string cmd = command_vm["command"].as<string>();
+
+      if (cmd == "account") {
+        auto account_file = getAccountFile();
+        if (fileExist(account_file)) {
+          cout << "Account file exist at " << account_file << endl;
+          return 1;
+        }
+
+        auto account = dev::KeyPair::create();
+        auto json = getAccountJson(account);
+        if (!writeFile(account_file, json)) {
+          return 1;
+        }
+
+        cout << "Generate account file at " << account_file << endl;
+        cout << "Please save the secret key at a safe place. Account secret key: " << toHex(account.secret().ref())
+             << endl;
+        cout << "Account public key: " << account.pub() << endl;
+        cout << "Account address: " << account.address() << endl;
+
+        return 1;
+      }
+
+      if (cmd == "account-from-key") {
+        auto account_file = getAccountFile();
```

Comment:
do we support multicommand ? if not why do you check all command options even if the previous one was specified. If you used some function like <enum Command> getCommand(std::string) you could handle there also invalid commands and use here switch...